File: server/src/main/java/org/learning/server/entity/OrgNode.java
package org.learning.server.entity;


import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * A node of Org type, like org, department, team
 */
@Entity
public class OrgNode {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer id;
    private String name;
    private String description;
    private Integer parentNode;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getParentNode() {
        return parentNode;
    }

    public void setParentNode(Integer parentNode) {
        this.parentNode = parentNode;
    }
}

File: server/src/main/java/org/learning/server/entity/OrgNode.kt
package org.learning.server.entity

import javax.persistence.*

/**
 * 表示一个架构节点，当节点没有parent时，表示其为组织（organization），否则是一个部门（department），为树型的架构
 */
@Entity
class OrgNode {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = -1
    var name: String = ""
    var description: String = ""
//    @ManyToOne
//    var parent: OrgNode = OrgNode()
    /**
     * 父级节点，这里需要手动管理。
     */
    var parentId: Int? = null
    /**
     * 表示这个结构节点是否公开（仅对顶级节点有效）
     */
    var public: Boolean = false
}